src/App.tsx

```diff
@@ -15,71 +16,104 @@ export const goods = [
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+// interface Props {
+//   goodsList: string[];
+// }
+
+interface State {
+  selectedGood: string;
+}
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGood: 'Jam',
+  };
+
+  handleGoodSelection = (_event: React.MouseEvent, good: string) => {
+    this.setState({
+      selectedGood: good,
+    });
+  };
+
+  handleGoodRemoval = () => {
+    this.setState({
+      selectedGood: '',
+    });
+  };
+
+  render() {
+    const { selectedGood } = this.state;
+
+    return (
+
+      <main className="section container">
+
+        {selectedGood
+          ? (
+            <h1 className="title is-flex is-align-items-center">
+              {`${selectedGood} is selected`}
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={this.handleGoodRemoval}
+              />
+            </h1>
+
+          )
+          : <h1 className="title">No goods selected</h1>}
+
+        <table className="table">
+          <tbody>
+            {goods.map(good => {
+              let isSelected = false;
+
+              if (good === selectedGood) {
+                isSelected = true;
+              }
+
+              const className = cn({
+                'has-background-success-light': isSelected,
+              });
+
+              return (
+                <tr key={good} data-cy="Good" className={className}>
+                  <td>
+                    {isSelected
```

Comment:
you can do it this way and avoid this part of code above:
```
       let isSelected = false;

       if (good === selectedGood) {
         isSelected = true;
       }
```
```suggestion
   {good === selectedGood
```